src/scrabble.js

```diff
@@ -1,5 +1,97 @@
-function scrabble() {
-  // write code here
+const eachLetterValues = {
+  A: 1,
+  E: 1,
+  I: 1,
+  O: 1,
+  U: 1,
+  L: 1,
+  N: 1,
+  R: 1,
+  S: 1,
+  T: 1,
+  D: 2,
+  G: 2,
+  B: 3,
+  C: 3,
+  M: 3,
+  P: 3,
+  F: 4,
+  H: 4,
+  V: 4,
+  W: 4,
+  Y: 4,
+  K: 5,
+  J: 8,
+  X: 8,
+  Q: 10,
+  Z: 10
 }
 
+// Function to calculate the score for a letter with double or triple modifiers
+function calculateLetterScore(letter, modifier) {
+  const letterValue = eachLetterValues[letter] || 0
+  if (modifier === '{}') {
+    return letterValue * 2
+  } else if (modifier === '[]') {
+    return letterValue * 3
+  } else {
+    return letterValue
+  }
+}
+
+// Function to calculate the score for a word with double or triple word modifiers
+function calculateWordScore(word, modifier) {
+  let wordScore = 0
+  for (let i = 0; i < word.length; i++) {
+    const letter = word[i]
+    const letterModifier =
+      word[i - 1] === '{' ? '{}' : word[i - 1] === '[' ? '[]' : ''
+    wordScore += calculateLetterScore(letter, letterModifier)
+  }
+  if (modifier === '{}') {
+    return wordScore * 2
+  } else if (modifier === '[]') {
+    return wordScore * 3
+  } else {
+    return wordScore
+  }
+}
+
+function scrabble(word) {
+  if (!word) return 0
```

Comment:
Not sure what this is actually doing? Can you clarify?

Reply:
The line if (!word) return 0; is a conditional statement that checks the value of the word variable and returns 0 under certain conditions